Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1401,6 +1401,17 @@ extension BrowserViewController: WKNavigationDelegate {
                     return
                 }
 
+                if let readerMode = tab.getHelper(name: ReaderMode.name()) as? ReaderMode {
+                    self.urlBar.updateReaderModeState(readerMode.state)
```

Comment:
Hm, the bug was about the reader mode bar, so I wasn't expecting to see changes to the URL bar/reader icon. Why is this change needed?